### PR TITLE
Add `WriteToSecret` method to `Reconciler` interface

### PR DESCRIPTION
**Pull request by ndbhat**
Description of changes:
WriteToSecret writes a string value to a Secret given the namespace, name, and key of the Secret

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/runtime/reconciler.go [hunk 1]
```diff
@@ -152,6 +153,43 @@ func (r *reconciler) SecretValueFromReference(
...
+	nsn := types.NamespacedName{
+		Name: name,
+	}
+	nsn.Namespace = namespace
+
+	secret := &corev1.Secret{}
+	err := r.apiReader.Get(ctx, nsn, secret)
+	if err != nil {
+		return ackerr.SecretNotFound
+	}
```
Comment (a-hilaly) [inline pkg/runtime/reconciler.go:176]: I think we don't need to do the `Get` here. Patch will fail anyways if the secret doesn't exist. Can you test this function with no `Get` calls?

Comment (ndbhat) [reply pkg/runtime/reconciler.go:176]: `Get` retrieves the secret from the given namespace and name so I think we need to call it before Patch. I followed the field export WriteToSecret method [https://github.com/aws-controllers-k8s/runtime/blob/e5668082edc3bf40d7d2d6bdcf43e0189426d528/pkg/runtime/field_export_reconciler.go#L336C1-L382C1](https://github.com/aws-controllers-k8s/runtime/blob/e5668082edc3bf40d7d2d6bdcf43e0189426d528/pkg/runtime/field_export_reconciler.go#L336C1-L382C1)

File: go.mod
Comment (a-hilaly) [inline go.mod:1]: Let's not include this change

File: pkg/runtime/reconciler.go (hunk 1 above)
Comment (a-hilaly) [reply pkg/runtime/reconciler.go:166]: Makes sense, thank you!

Comment (a-hilaly) [reply pkg/runtime/reconciler.go:166]: One last ask, can we reuse this function in fieldExportReconciler?

Comment (a-hilaly) [reply pkg/runtime/reconciler.go:166]: Discussed with Neha off-line we will address this in a seperate PR before releasing runtime 0.34.0

**Review by a-hilaly — APPROVED** [review]
👍 
/lgtm

**Pull request merged**